### PR TITLE
Switch to Rollup

### PR DESCRIPTION
**Pull request by IanMitchell**
Similar to #74 - this creates a bundle that works for Node.js CJS and the browser, but it goes about it a different way. Rather than using Webpack as a bundler it switches to rollup.

I also removed the `dotenv` functionality in favor of requiring a key in the config - the dotenv stuff was undocumented but it probably warrants a major version bump if this PR ships. This helps reduce the size significantly for the browser.

I added an API key in the travis file so that the test suite can run - that said, it looks like API limits restrict this significantly. I thought about modifying the tests to check stubbed responses from the API but it doesn't seem like there is value there - happy to take guidance on ways to update the test suite if you want!

Only other question I have is on the batch API - it looks like this doesn't exist, was it removed? Calling that code now throws an exception on my machine.

### CODE REVIEW

**Comment by IanMitchell** [comment]
I believe the travis build fails since this is a forked PR, so for security reasons it isn't loading the new API key

**Comment by zackurben** [comment]
Thanks for the contribution @IanMitchell, give me a day or two to review this! 🎉

**Comment by zackurben** [comment]
It looks like there are some legit test failures due to underlying service changes. I'm going to pull this down and check em out.

**Comment by IanMitchell** [comment]
Holler if you need help! Happy to spend more time on this if you want

File: .env.example [hunk 1]
```diff
@@ -1 +1 @@
-AV_KEY=demo
\ No newline at end of file
+API_KEY=demo
```
Comment (zackurben) [inline .env.example]: Can we revert this API key rename? I don't want to break env configs 
```suggestion
AV_KEY=demo
```

File: index.js [hunk 2]
```diff
@@ -1,37 +1,19 @@
...
-  ['key'].forEach(prop => {
-    if (config[prop] === undefined) {
-      errors.push(prop);
-    }
-  });
-  if (errors.length) {
-    throw new Error(`Missing Alpha Vantage config settings: ${errors.join(', ')}`);
+  // Check for API Key
+  if (config.key === undefined) {
+    throw new Error('Missing Alpha Vantage config settings: key');
```
Comment (zackurben) [inline index.js]: This change breaks [the following test](https://github.com/IanMitchell/alphavantage/blob/2a78e1c82f5b0ce2406d4e31a129b69303511139/test/init.js#L27-L36).

I'm fine with keeping this change, but we need to remove the test.

File: test/crypto.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import Alpha from '../';
 import delay from 'delay';
-const alpha = Alpha();
+import Alpha from '../';
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/crypto.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: test/data.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import Alpha from '../';
 import delay from 'delay';
-const alpha = Alpha();
+import Alpha from '../';
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/data.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: test/forex.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import Alpha from '../';
 import delay from 'delay';
-const alpha = Alpha();
+import Alpha from '../';
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/forex.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: test/performance.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import Alpha from '../';
 import delay from 'delay';
-const alpha = Alpha();
+import Alpha from '../';
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/performance.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: test/technical.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import Alpha from '../';
 import delay from 'delay';
-const alpha = Alpha();
+import Alpha from '../';
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/technical.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: test/util.js [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 jest.mock('cross-fetch');
 import Alpha from '../';
-const alpha = Alpha();
+const alpha = Alpha({ key: process.env.API_KEY });
```
Comment (zackurben) [inline test/util.js]: Lets keep the old env name

```suggestion
const alpha = Alpha({ key: process.env.AV_KEY });
```

File: .travis.yml [hunk 9]
```diff
@@ -1,6 +1,8 @@
...
 node_js:
-  - "lts/*"
-
+- lts/*
 after_script:
-  - npm run coveralls
+- npm run coveralls
+env:
+  global:
+    secure: S/C5661Lm9LD/NVXg4h6lA8Fy2y3suAprXVqz9eycCM0Vspb5f7XClvS+8wdizC2Q63rqhGJe6gkeNiJ+KUsAPLrk8gIKvg7alNJS9ebyUZaHfxy0HYjBj2eDtVY8ISJXoaBP7dFWSjwjw6dX8HVKIIjwidvF3EEjTF0jue4q54xTKERcRXZBwIV/PnupgLjSFgjvkUSgAD+SK0tLfqoExI9rHFOqugFh1l3u6p+WPV7GbQWktOFyrYoC0SZzLPUBLz5KBin/AwnHFFHg7nabObSEOSvcqH73Gp6vu4dodjZONGsI90K5KRwHhZ4SSt28f2L0XgEarFlhN1/COtEr2ceJN1HgpbPy/R8D11vIGz0fvIKJcPuONTv5XWHYStkyGpKxkbR5Dbfs9xz4PZlBZGD8o1OQGkWQMZoLGDh5f3F/L1Ntz9YyKkST7IIY35hy578Dzmm3ykRgqfxwucj1b2Cxhr/mKvyib8AUCcCEYDwYtPw8++x2AXdQR/WJ3+M9y6GzJmGVfmwdJbcc2uAaSbXrLnQhVGCFNTHnxSnjb+oxwvObixsDzROlcOfKrHs2SUk9cuQHYqnAHuAWeLIoLzn1P3QXWvL16fVUsUn3VVAdrkBioR1Btb4hGaZF0CuXUFh0cMBrqXImmY/y26Xd2Qvo+qyybjlbFOogthu5ko=
```
Comment (zackurben) [inline .travis.yml]: What env is this? is this your personal AV API key in travis ci?

File: index.js [hunk 10]
```diff
@@ -44,6 +26,6 @@ export default (config = {}) => {
     forex: Forex(config),
     crypto: Crypto(config),
     technical: Technical(config),
-    performance: Performance(config)
+    performance: Performance(config),
```
Comment (zackurben) [inline index.js]: Please remove this trailing comma, the lint script will do this: `npm run lint`

File: babel.config.js [hunk 11]
```diff
@@ -1,13 +1,12 @@
...
-          node: 'current'
-        }
-      }
-    ]
-  ]
+          node: 'current',
+        },
+      },
+    ],
+  ],
```
Comment (zackurben) [inline babel.config.js]: Please remote the trailing commas here.

**Comment by IanMitchell** [comment]
Yeah I’ll definitely remove it and update the PR!

File: .travis.yml (hunk 9 above)
Comment (IanMitchell) [reply .travis.yml]: This is a new one I generated for Travis CI - it's encoded so shouldn't be accessible except for the platform

File: index.js (hunk 10 above)
Comment (IanMitchell) [reply index.js]: Prettier updated and has this behavior by default now - I'll add the option to turn off this feature.

Comment (IanMitchell) [reply index.js]: Wait, I'm wrong - the lint write rules don't target these files. I extracted the files into a .prettierrc so this shouldn't happen again 😄

**Pull request merged**